lib/engine/ability/train_discount.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Ability
+    class TrainDiscount < Base
+      attr_reader :discount, :absolute, :trains
```

Comment:
new line

Reply:
ok i was confused, i thought discount absolute meant that the new price is X

maybe we should just make discount an integer subtraction and a float for percentage?

so you can do discount 50 for minus 50$ or 0.5 for 50% off

Reply:
If discount > 1 the new price will be old_price - discount
otherwise the new price is old_price - discount*old_price, ie old_price*(1-discount)

Fixed